core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1829,7 +1901,63 @@ private[spark] class BlockManager(
     data.dispose()
   }
 
+  /**
+   * Class to handle block manager decommissioning retries
+   * It creates a Thread to retry offloading all RDD cache blocks
+   */
+  private class BlockManagerDecommissionManager(conf: SparkConf) {
+    @volatile private var stopped = false
+    private val sleepInterval = conf.get(
+      config.STORAGE_DECOMMISSION_REPLICATION_REATTEMPT_INTERVAL)
+
+    private val blockReplicationThread = new Thread {
+      override def run(): Unit = {
+        var failures = 0
+        try {
+          while (blockManagerDecommissioning &&
+            !stopped &&
+            !Thread.interrupted() &&
+            failures < 20) {
+            logDebug("Attempting to replicate all cached RDD blocks")
+            try {
+              decommissionRddCacheBlocks()
+            } catch {
+              case e: InterruptedException =>
```

Comment:
Can this explicit catching of InterruptedException `case e: InterruptedException =>` be removed? InterruptException will not be caught by `case NonFatal(t)` here so it will automatically be thrown back?

Reply:
That's probably ok